api/v1alpha1/acl_types.go

```diff
@@ -0,0 +1,24 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// +kubebuilder:validation:XValidation:rule="(has(self.allow) || has(self.deny))",message="one of allow or deny must be specified"
```

Comment:
What will happen if both allow and deny are specified? Should this be allowed?

Reply:
actually that was same question that I asked myself. However, there is usecase for that.

Lets say that I have following config:

```
  acl:
    allow:
    - cidr: "1.2.3.0/24"
    deny:
    - cidr: "1.2.3.120/32"
```

so this configuration means that it will accept all traffic from 1.2.3.0/24. However, it will block that single ip address 1.2.3.120. So it can be used for "allow big chunk and deny smaller pieces of it"